Improve Error Message for Dist Autograd Context Cleanup Failure

osalpekar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37255 Improve Error Message for Dist Autograd Context Cleanup Failure**

Improved error message logged when Distributed Autograd Context cleanup fails - added node information and underlying error. The previous error message also assumed that the cause of the error was due to too many RPC's failing, but this is not necessarily the case.

Differential Revision: [D20950664](https://our.internmc.facebook.com/intern/diff/D20950664/)

osalpekar committed:
Improve Error Message for Dist Autograd Context Cleanup Failure

Improved error message logged when Distributed Autograd Context cleanup fails - added node information and underlying error. The previous error message also assumed that the cause of the error was due to too many RPC's failing, but this is not necessarily the case.

Differential Revision: [D20950664](https://our.internmc.facebook.com/intern/diff/D20950664/)

[ghstack-poisoned]

osalpekar committed:
Update on "Improve Error Message for Dist Autograd Context Cleanup Failure"


Improved error message logged when Distributed Autograd Context cleanup fails - added node information and underlying error. The previous error message also assumed that the cause of the error was due to too many RPC's failing, but this is not necessarily the case.

Differential Revision: [D20950664](https://our.internmc.facebook.com/intern/diff/D20950664/)

[ghstack-poisoned]

mrshenli commented on torch/csrc/distributed/autograd/context/container.cpp:
```diff
@@ -237,12 +237,13 @@ void DistAutogradContainer::sendReleaseContextRpc(
+              "Could not release Dist Autograd Context on node ",
+              worker_id,
+              ": ",
+              error->what());
```
where does this error come from?

osalpekar replied:
This is when cleaning up the autograd contexts on neighboring nodes fails. It may happen if there's an ungraceful shutdown (as in `test_backward_node_failure`) or if all the ReleaseContextRpc's fail.

osalpekar replied:
Changing to `cleanupFuture.error->what()`

osalpekar committed:
Update on "Improve Error Message for Dist Autograd Context Cleanup Failure"


Improved error message logged when Distributed Autograd Context cleanup fails - added node information and underlying error. The previous error message also assumed that the cause of the error was due to too many RPC's failing, but this is not necessarily the case.

Differential Revision: [D20950664](https://our.internmc.facebook.com/intern/diff/D20950664/)

[ghstack-poisoned]
